parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -116,48 +122,48 @@ private boolean eval(
         }
       }
 
-      return ExpressionVisitors.visitEvaluator(expr, this);
+      return ExpressionVisitors.visit(expr, this);
     }
 
     @Override
-    public Boolean alwaysTrue() {
-      return ROWS_MIGHT_MATCH; // all rows match
+    public Expression alwaysTrue() {
+      return ROWS_ALL_MATCH; // all rows match
     }
 
     @Override
-    public Boolean alwaysFalse() {
+    public Expression alwaysFalse() {
       return ROWS_CANNOT_MATCH; // all rows fail
     }
 
     @Override
-    public Boolean not(Boolean result) {
-      return !result;
+    public Expression not(Expression result) {
+      throw new UnsupportedOperationException("This path shouldn't be reached.");
```

Comment:
Yes, you're right, it should already been rewritten. 

It's just when I was refactoring this part, I found that the dict filter and metric filter handle `not` inconsistently compared to the bloom filter. The bloom filter will directly throw an `UnsupportedOperationException` because it cannot evaluate `not`. In fact, the dict filter and metric filter also cannot handle not since they can only conclude `ROWS_CANNOT_MATCH` or `ROWS_MIGHT_MATCH`. So, I made their handling of `not` consistent with the bloom filter. Anyway, this should not be reached.

Is this ok? I can revert this if not.